istio/discovery.go

```diff
@@ -754,11 +777,25 @@ func (in *Discovery) Mesh(ctx context.Context) (*models.Mesh, error) {
 		}
 	}
 
+	// add set the NamespaceMap
+	mesh.NamespaceMap = map[string]*models.ControlPlane{}
+	for _, cp := range controlPlanes {
+		for _, mn := range cp.ManagedNamespaces {
+			key := Key(mn.Cluster, mn.Name)
+			mesh.NamespaceMap[key] = cp
+		}
+	}
+
 	in.kialiCache.SetMesh(mesh)
 
 	return mesh, nil
 }
 
+// key returns "cluster:namespace"
+func Key(cluster, namespace string) string {
+	return cluster + ":" + namespace
+}
```

Comment:
`Key` is not specific. Key for what?

Reply:
+1, got distracted and didn't finish this. Sorry about that.